### ISSUE TITLE
validate all models, keep best

### ISSUE THREAD

**Issue by lacava** [lacava/feat#250]
at the moment the strategy for model validation / early stalling is to take the "middlest" model in the population and test its validation score. if it is not improving, the stall count increases. meanwhile, the best model is stored based on training data, and the final best model is chosen from either the final population or archive at the end of the run, based on validation loss. 

this strategy is fairly lightweight because it only validates one model per generation, but risks losing the best validating model found during the run. so my suggest is to instead calculate the validation loss of all models as a matter of course, and keep the best individual based on validation loss found during the run, breaking ties with complexity. the stall count could also depend on improvements to this best validation loss, or to some other statistic of the population's validation performance.

**Comment by lacava** [comment]
this was merged into master in commit 5692bce9
